designer/server/src/main/scala/pl/touk/nussknacker/ui/api/description/MigrationApiEndpoints.scala

```diff
@@ -98,6 +115,88 @@ class MigrationApiEndpoints(auth: EndpointInput[AuthCredentials]) extends BaseEn
 object MigrationApiEndpoints {
 
   object Dtos {
+    sealed trait MigrationError
+
+    object MigrationError {
+      final case class Validation(errors: ValidationErrors)                                extends MigrationError
+      final case class MigrationToArchived(processName: ProcessName, environment: String)  extends MigrationError
+      final case class Unauthorized(user: LoggedUser)                                      extends MigrationError
+      final case class MigrationApiAdapter(apiAdapterServiceError: ApiAdapterServiceError) extends MigrationError
+
+      implicit val validationErrorCodec: Codec[String, Validation, CodecFormat.TextPlain] =
+        Codec.string.map(
+          Mapping.from[String, Validation](deserializationException)(validationError => {
+            val errors = validationError.errors
+
+            val messages = errors.globalErrors.map(_.error.message) ++
+              errors.processPropertiesErrors.map(_.message) ++ errors.invalidNodes.map { case (node, nerror) =>
+                s"$node - ${nerror.map(_.message).mkString(", ")}"
+              }
+            s"Cannot migrate, following errors occurred: ${messages.mkString(", ")}"
+          })
+        )
+
+      final case class RemoteEnvironmentCommunicationError(message: String) extends MigrationError
+
+      final case class Unknown(message: String) extends MigrationError
```

Comment:
I don't think we need it. It should be simply converted to HTTP 500. So, in the place where we instantiate the Unknown, there should be an exception thrown.